### PR TITLE
Refactor SKonsole Commands and Improve Code Consistency

### LINKED ISSUE #6: Refactor commands out of program.cs

### PR DESCRIPTION
**Pull request by lemillermicrosoft**
Resolves #6.

Summary:
This pull request introduces a series of changes aimed at refactoring the SKonsole codebase, improving code organization, maintainability, and consistency. The main changes involve updating Java formatting settings, refactoring SKonsole commands, improving exception handling, and adding new settings and recommendations for Visual Studio Code.

Detailed List of Changes:
1. Updated c# formatting settings in the `.editorconfig` file, such as spacing rules for various operators, parentheses, and braces.
2. Refactored the `SKonsole` commands in the `ConfigCommand.cs` and `ConfigurationProvider.cs` files, including the use of `this` keyword for instance members, updating class declarations to be sealed, and refining exception handling.
3. Improved exception handling in `ConfigurationProvider.cs` by handling specific exceptions: `JsonException`, `NotSupportedException`, and `ArgumentNullException`.
4. Removed unused `using System.CommandLine.Builder;` directive and wrapped `loggerFactory` in a `using` statement in `Program.cs`.
5. Added braces to single-line `if` statements in `Program.cs` for improved readability.
6. Changed class declarations from `internal class` to `internal sealed class` in `EmailSkill.cs`, `GitSkill.cs`, and `WriterSkill.cs`.
7. Added a new `.vscode/settings.json` file with settings for formatting, bracket pair colorization, file exclusions, and custom words for the spell checker.
8. Added a new `extensions.json` file with recommendations for Visual Studio Code extensions, including C# support, Semantic Kernel, and Code Spell Checker.

These changes will help ensure that the project adheres to best practices and maintains a high level of code quality, consistency, and readability.

### CODE REVIEW

**Comment by lemillermicrosoft** [comment]
FYI @xbotter

File: apps/SKonsole/Commands/PromptChatCommand.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            }
+
+            history += $"{botMessageFormatted}Human: {userMessage}\nAI:";
+            contextVariables.Set("history", history);
+
+            botMessage = await kernel.RunAsync(contextVariables, chatFunction);
+        }
+    }
+
+    private static string ReadMutiLineInput()
```
Comment (xbotter) [inline apps/SKonsole/Commands/PromptChatCommand.cs:93]: fix typo
```suggestion
    private static string ReadMultiLineInput()
```

File: apps/SKonsole/Commands/PromptChatCommand.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        var botMessage = await kernel.RunAsync(contextVariables, chatFunction);
+        var userMessage = string.Empty;
+
+        while (userMessage != "exit")
+        {
+            var botMessageFormatted = "\nAI: " + botMessage.ToString() + "\n";
+            (logger ?? kernel.Logger).LogInformation("{botMessage}", botMessageFormatted);
+            (logger ?? kernel.Logger).LogInformation(">>>");
+
+            userMessage = ReadMutiLineInput();
```
Comment (xbotter) [inline apps/SKonsole/Commands/PromptChatCommand.cs:80]: fix typo
```suggestion
            userMessage = ReadMultiLineInput();
```

**Review by xbotter — COMMENTED** [review]
fix a typo

**Pull request merged**